src/views/UserDetails.vue

```diff
@@ -0,0 +1,236 @@
+              </ion-button>
+              <ion-button v-else @click="updateContact('email')" slot="end" fill="clear">
+                <ion-icon slot="icon-only" :icon="addCircleOutline" />
+              </ion-button>
+            </ion-item>
+            <ion-item>
+              <ion-icon :icon="callOutline" slot="start" />
+              <ion-label>{{ selected.contactNumber ? selected.contactNumber : translate('Phone number') }}</ion-label>
+              <ion-button v-if="selected.contactNumber" slot="end" fill="clear" color="medium" @click="openContactActionsPopover($event, 'phoneNumber', selected.contactNumber)">
+                <ion-icon slot="icon-only" :icon="ellipsisVerticalOutline" />
+              </ion-button>
+              <ion-button v-else @click="updateContact('phoneNumber')" slot="end" fill="clear">
+                <ion-icon slot="icon-only" :icon="addCircleOutline" />
+              </ion-button>
+            </ion-item>
+            <ion-item>
+              <ion-icon :icon="businessOutline" slot="start" />
+              <ion-label>{{ selected.externalId ? selected.externalId : translate('External ID') }}</ion-label>
+              <ion-button v-if="selected.externalId" slot="end" fill="clear" color="medium" @click="openContactActionsPopover($event, 'externalId', selected.externalId)">
+                <ion-icon slot="icon-only" :icon="ellipsisVerticalOutline" />
+              </ion-button>
+              <ion-button v-else @click="updateContact('externalId')" slot="end" fill="clear">
+                <ion-icon slot="icon-only" :icon="addCircleOutline" />
+              </ion-button>
+            </ion-item>
+          </ion-card>
+        </section>
+      </main>
+    </ion-content>
+  </ion-page>
+</template>
+<script lang="ts">
+import {
+  alertController,
+  IonAvatar,
+  IonButton,
+  IonCard,
+  IonCardHeader,
+  IonCardTitle,
+  IonContent,
+  IonHeader,
+  IonIcon,
+  IonItem,
+  IonLabel,
+  IonMenuButton,
+  IonPage,
+  IonTitle,
+  IonToggle,
+  IonToolbar,
+  popoverController,
+} from "@ionic/vue";
+import { defineComponent } from "vue";
+import { useRouter } from 'vue-router';
+import { mapGetters, useStore } from 'vuex'
+import {
+  addCircleOutline,
+  businessOutline,
+  callOutline,
+  ellipsisVerticalOutline,
+  mailOutline,
+  warningOutline
+} from 'ionicons/icons';
+import Image from '@/components/Image.vue'
+import { translate } from '@hotwax/dxp-components';
+import ContactActionsPopover from '@/components/ContactActionsPopover.vue'
+
+export default defineComponent({
+  name: "User Details",
+  components: {
+    Image,
+    IonAvatar,
+    IonButton,
+    IonCard,
+    IonCardHeader,
+    IonCardTitle,
+    IonContent,
+    IonHeader,
+    IonIcon,
+    IonItem,
+    IonLabel,
+    IonMenuButton,
+    IonPage,
+    IonTitle,
+    IonToggle,
+    IonToolbar,
+  },
+  computed: {
+    ...mapGetters({
+      selected: 'user/getSelectedUser'
```

Comment:
We should call it selectedUser, or currentUser or just user, only `selected` is not self-explanatory.